src/common/runtime/helper/wrap_for_worker.ts

```diff
@@ -10,25 +11,33 @@ import { setupWorkerEnvironment, WorkerTestRunRequest } from './utils_worker.js'
 export function wrapTestGroupForWorker(g: TestGroup<Fixture>) {
   self.onmessage = async (ev: MessageEvent) => {
     const { query, expectations, ctsOptions } = ev.data as WorkerTestRunRequest;
+    try {
+      const log = setupWorkerEnvironment(ctsOptions);
 
-    const log = setupWorkerEnvironment(ctsOptions);
-
-    const testQuery = parseQuery(query);
-    assert(testQuery instanceof TestQuerySingleCase);
-    let testcase = null;
-    for (const t of g.iterate()) {
-      if (comparePaths(t.testPath, testQuery.testPathParts) !== Ordering.Equal) {
-        continue;
-      }
-      for (const c of t.iterate(testQuery.params)) {
-        if (comparePublicParamsPaths(c.id.params, testQuery.params) === Ordering.Equal) {
-          testcase = c;
+      const testQuery = parseQuery(query);
+      assert(testQuery instanceof TestQuerySingleCase);
+      let testcase = null;
+      for (const t of g.iterate()) {
+        if (comparePaths(t.testPath, testQuery.testPathParts) !== Ordering.Equal) {
+          continue;
+        }
+        for (const c of t.iterate(testQuery.params)) {
+          if (comparePublicParamsPaths(c.id.params, testQuery.params) === Ordering.Equal) {
+            testcase = c;
+          }
         }
       }
+      assert(!!testcase, 'testcase not found');
+      const [rec, result] = log.record(query);
+      await testcase.run(rec, testQuery, expectations);
+
+      ev.source?.postMessage({ query, result });
+    } catch (thrown) {
+      const ex = thrown instanceof Error ? thrown : new Error(`${thrown}`);
```

Comment:
Out of curiosity, when is `thrown` not an Error?

Reply:
It probably won't ever be, but it's theoretically possible, so TypeScript makes us handle it. Since this is error handling code I prefer to handle it than to just type-cast from `unknown` to `Error`. Otherwise a new exception could be thrown and lost the same way `thrown` was lost before.